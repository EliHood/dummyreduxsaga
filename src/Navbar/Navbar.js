import React from 'react';
import {
  BrowserRouter as Router, Route, Link, Switch,
} from 'react-router-dom';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import Home from '../containers/home';

const Navbar = () => (
  <Router>
    <AppBar position="static">
      <Toolbar>
        <Grid justify="space-between" container>
          <Typography variant="h6" style={{ color: '#fff' }}>
                   Boilerplate
          </Typography>
          {/* if is authenticated, will render authlinks
                    if not will render guest links
                */}
          <Grid item>
            <Button align="right">
              <Link style={{ color: '#fff', textDecoration: 'none' }} underline="none" to="/">
                            Home
              </Link>

            </Button>
          </Grid>
        </Grid>
      </Toolbar>
    </AppBar>

    <Route exact path="/" component={Home} />

  </Router>
);
export default Navbar;
