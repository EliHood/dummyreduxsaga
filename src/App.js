import React, { Component } from 'react';
import { Provider } from 'react-redux';
import Grid from '@material-ui/core/Grid';
import Navbar from './Navbar/Navbar';
import store from './store';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <Provider store={store}>
          <Navbar />
          <Grid container />
        </Provider>
      </div>
    );
  }
}
export default App;
