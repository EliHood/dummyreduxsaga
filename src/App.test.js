import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './App';
import Navbar from './Navbar/Navbar';

describe('Should render App component', () => {
  it('should render App component', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find('div')).toHaveLength(1);
  });

  it('should contain Navbar', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find(Navbar)).toHaveLength(1);
  });


  it('should snapshot Navbar', () => {
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
