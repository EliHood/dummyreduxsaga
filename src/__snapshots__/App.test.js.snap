// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render App component should snapshot Navbar 1`] = `
<div
  className="App"
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Navbar />
    <WithStyles(ForwardRef(Grid))
      container={true}
    />
  </Provider>
</div>
`;
